src/object/srv_obj.c

```diff
@@ -4154,13 +4226,29 @@ ds_obj_dtx_leader_ult(void *arg)
 	D_ASSERT(dcsh->dcsh_epoch.oe_value != 0);
 	D_ASSERT(dcsh->dcsh_epoch.oe_value != DAOS_EPOCH_MAX);
 
+	dcde = ds_obj_cpd_get_dcde(dca->dca_rpc, dca->dca_idx, 0);
+	dcsrs = ds_obj_cpd_get_dcsr(dca->dca_rpc, dca->dca_idx);
+	tgts = ds_obj_cpd_get_tgts(dca->dca_rpc, dca->dca_idx);
+	req_cnt = ds_obj_cpd_get_dcsr_cnt(dca->dca_rpc, dca->dca_idx);
+	tgt_cnt = ds_obj_cpd_get_tgt_cnt(dca->dca_rpc, dca->dca_idx);
+
+	if (dcde == NULL || dcsrs == NULL || tgts == NULL ||
+	    req_cnt < 0 || tgt_cnt < 0)
+		D_GOTO(out, rc = -DER_INVAL);
+
+	if (tgt_cnt <= 1 && dcde->dcde_write_cnt <= 1)
```

Comment:
could it be zero? just wondering why it's "<=1" not "==1"

Reply:
write_cnt can be zero, that for readonly TX.